### PR TITLE
sys: util: Add 64bit mask support

### PR DESCRIPTION
**Pull request by nirbaror**
While working with 64bit values, found these to be missing.

### CODE REVIEW

File: include/zephyr/sys/util.h [hunk 1]
```diff
@@ -61,6 +61,13 @@ extern "C" {
 #define GENMASK(h, l) \
 	(((~0UL) - (1UL << (l)) + 1) & (~0UL >> (BITS_PER_LONG - 1 - (h))))
 
+/**
+ *  * @brief Create a contiguous 64-bit bitmask starting at bit position @p l
```
Comment (mfischer) [inline include/zephyr/sys/util.h:65]: Please fix

Comment (nirbaror) [reply include/zephyr/sys/util.h]: Done, thanks!

File: include/zephyr/sys/util.h [hunk 2]
```diff
@@ -44,20 +44,30 @@ extern "C" {
 /** @brief Cast @p x, a signed integer, to a <tt>void*</tt>. */
 #define INT_TO_POINTER(x)  ((void *) (intptr_t) (x))
 
-#if !(defined(__CHAR_BIT__) && defined(__SIZEOF_LONG__))
+#if !(defined(__CHAR_BIT__) && defined(__SIZEOF_LONG__) && defined(__SIZEOF_LONG_LONG__))
```
Comment (andyross) [inline include/zephyr/sys/util.h:47]: This is going to cause new build errors if someone is building with a toolchain that doesn't define __SIZEOF_LONG_LONG__ (which AFAIK is a gcc-ism, not part of a C standard).  It's probably unlikely, but something to be aware of.

**Review by andyross — COMMENTED** [review]
Code looks fine to me.  I do note there's a stray merge commit in the series that will need to be squashed.

File: include/zephyr/sys/util.h (hunk 2 above)
Comment (nirbaror) [reply include/zephyr/sys/util.h:47]: These are the other options I see (if sizeof isn't defined)

#define GENMASK64(h, l) \
	(((~0ULL) - (1ULL << (l)) + 1) & (~0ULL >> (64 - 1 - (h))))

#define GENMASK64(h, l) \
    ((h) == 63 && (l) == 0) ? \
        ~0ULL : \
        (((1ULL << ((h)-(l)+1)) - 1) << (l))

**Review by andyross — APPROVED** [review]
Thanks

**Review by cfriedt — APPROVED** [review]

**Pull request merged**